src/celeritas/ext/detail/CelerOpticalPhysics.cc

```diff
@@ -0,0 +1,337 @@
+    // G4OpticalPhotons
+    G4OpticalPhoton::OpticalPhotonDefinition();
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Build list of available processes and models.
+ */
+void CelerOpticalPhysics::ConstructProcess()
+{
+    auto* process_manager
+        = G4OpticalPhoton::OpticalPhoton()->GetProcessManager();
+    CELER_ASSERT(process_manager);
+
+    // Add Optical Processes
+    // TODO: Celeritas will eventually implement these directly (no
+    // G4OpticalPhotons) so how to set up on "Celeritas-side"
+    auto absorption = std::make_unique<G4OpAbsorption>();
+    if (process_is_active(OpticalProcessType::absorption, options_))
+    {
+        process_manager->AddDiscreteProcess(absorption.release());
+        CELER_LOG(debug) << "Loaded Optical absorption with G4OpAbsorption "
+                            "process";
+    }
+
+    auto rayleigh = std::make_unique<G4OpRayleigh>();
+    if (process_is_active(OpticalProcessType::rayleigh, options_))
+    {
+        process_manager->AddDiscreteProcess(rayleigh.release());
+        CELER_LOG(debug)
+            << "Loaded Optical Rayleigh scattering with G4OpRayleigh "
+               "process";
+    }
+
+    auto mie = std::make_unique<G4OpMieHG>();
+    if (process_is_active(OpticalProcessType::mie_hg, options_))
+    {
+        process_manager->AddDiscreteProcess(mie.release());
+        CELER_LOG(debug) << "Loaded Optical Mie (Henyey-Greenstein phase "
+                            "function) scattering with G4OpMieHG "
+                            "process";
+    }
+
+    // NB: boundary is also used later on in loop over particles,
+    // though it's only ever applicable to G4OpticalPhotons
+    auto boundary = std::make_unique<G4OpBoundaryProcess>();
+#if G4VERSION_NUMBER < 1070
+    boundary->SetInvokeSD(options_.boundary.invoke_sd);
+#endif
+    if (process_is_active(OpticalProcessType::boundary, options_))
+    {
+        process_manager->AddDiscreteProcess(boundary.get());
+        CELER_LOG(debug)
+            << "Loaded Optical boundary process with G4OpBoundaryProcess "
+               "process";
+    }
+
+    auto wls = std::make_unique<G4OpWLS>();
+#if G4VERSION_NUMBER < 1070
+    wls->UseTimeProfile(to_cstring(options_.wavelength_shifting));
+#endif
+    if (process_is_active(OpticalProcessType::wavelength_shifting, options_))
+    {
+        process_manager->AddDiscreteProcess(wls.release());
+        CELER_LOG(debug) << "Loaded Optical wavelength shifting with G4OpWLS "
+                            "process";
+    }
+
+#if G4VERSION_NUMBER >= 1070
+    auto wls2 = std::make_unique<G4OpWLS2>();
+    if (process_is_active(OpticalProcessType::wavelength_shifting_2, options_))
+    {
+        process_manager->AddDiscreteProcess(wls2.release());
+        // I need to check how this differs from G4OpWLS...
+        CELER_LOG(debug)
+            << "Loaded Optical wavelength shifting V2 with G4OpWLS2 "
```

Comment:
Since WLS and WLS2 are identical, maybe

```suggestion
            << "Loaded second optical wavelength shifting with G4OpWLS2 process "
```